### DISCUSSION TITLE
Tasmota32 does not apply template after long time with no power

### DISCUSSION THREAD

**Discussion by msetina** [arendst/Tasmota#19048] [Q&A]
Using tasmota32 prebuild 13.0  () and applying basic settings and template. After a day off, turning the boad on I was surprised the template did not apply. I have an 8266 prepared with similar procedure and I am not seeing this issue.
This is from two command requested:
09:44:43.778 CMD: Backlog Template; Module; GPIO 255
09:44:43.815 MQT: stat/small_tasmota32/RESULT = {"NAME":"8relays8switches","GPIO":[230,0,231,0,229,164,0,0,0,0,0,0,0,0,161,162,0,225,224,163,0,227,160,226,0,0,0,0,228,166,167,165,0,0,0,0],"FLAG":0,"BASE":1}
09:44:44.047 MQT: stat/small_tasmota32/RESULT = {"Module":{"1":"ESP32-DevKit"}}
09:44:44.303 MQT: stat/small_tasmota32/RESULT = {"GPIO0":{"0":"None"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"0":"None"},"GPIO20":{"0":"None"},"GPIO21":{"0":"None"},"GPIO22":{"0":"None"},"GPIO23":{"0":"None"},"GPIO24":{"0":"None"},"GPIO25":{"0":"None"},"GPIO26":{"0":"None"},"GPIO27":{"0":"None"},"GPIO32":{"0":"None"},"GPIO33":{"0":"None"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO37":{"0":"None"},"GPIO38":{"0":"None"},"GPIO39":{"0":"None"}}
09:45:56.928 CMD: Status0
09:45:56.965 MQT: stat/small_tasmota32/STATUS0 = {"Status":{"Module":1,"DeviceName":"ESP32 enota","FriendlyName":["rele1"],"Topic":"small_tasmota32","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0},"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32.bin","RestartReason":"Software reset CPU","Uptime":"0T00:02:09","StartupUTC":"2023-07-06T08:43:47","Sleep":50,"CfgHolder":4617,"BootCount":27,"BCResetTime":"2023-06-29T13:38:57","SaveCount":65},"StatusFWR":{"Version":"13.0.0(tasmota32)","BuildDateTime":"2023-06-26T10:54:38","Core":"2_0_10","SDK":"4.4.5.230614","CpuFrequency":160,"Hardware":"ESP32-D0WD-V3 rev.3","CR":"548/699"},"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Borealis",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000088","00006000","00004000","00000000"]},"StatusMEM":{"ProgramSize":1870,"Free":1009,"Heap":113,"StackLowMark":3,"PsrMax":0,"PsrFree":0,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"16405E","FlashFrequency":40,"FlashMode":"DIO","Features":["00000809","9F9AD7DF","0015A001","B7F7BFCF","05DA9BC0","E0360DC7","480840D2","20200000","D4BC482D","000A80B1"],"Drivers":"1,2,3,4,5,7,8,9,10,11,12,14,16,17,20,21,24,26,27,29,34,35,38,50,52,59,60,62,63,66,67,68,82,86,87,88","Sensors":"1,2,3,5,6,7,8,9,10,11,12,13,14,15,17,18,19,20,21,22,26,31,34,37,39,40,42,43,45,51,52,55,56,58,59,64,66,67,74,85,92,95,98,103,105,127","I2CDriver":"7,8,9,10,11,12,13,14,15,17,18,20,24,29,31,36,41,42,44,46,48,58,62,65,69,76,77"},"StatusNET":{"Hostname":"small-tasmota32-4864","IPAddress":"192.168.44.31","Gateway":"192.168.44.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.1","DNSServer2":"0.0.0.0","Mac":"7C:87:CE:30:33:00","IP6Global":"","IP6Local":"fe80::7e87:ceff:fe30:3300","Ethernet":{"Hostname":"","IPAddress":"0.0.0.0","Gateway":"0.0.0.0","Subnetmask":"0.0.0.0","DNSServer1":"192.168.1.1","DNSServer2":"0.0.0.0","Mac":"00:00:00:00:00:00","IP6Global":"","IP6Local":""},"Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":16.0},"StatusMQT":{"MqttHost":"192.168.1.66","MqttPort":1883,"MqttClientMask":"big_relay_box","MqttClient":"big_relay_box","MqttUser":"transmota","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4},"StatusTIM":{"UTC":"2023-07-06T08:45:56","Local":"2023-07-06T09:45:56","StartDST":"2023-03-26T02:00:00","EndDST":"2023-10-29T03:00:00","Timezone":"+01:00","Sunrise":"04:54","Sunset":"20:55"},"StatusSNS":{"Time":"2023-07-06T09:45:56"},"StatusSTS":{"Time":"2023-07-06T09:45:56","Uptime":"0T00:02:09","UptimeSec":129,"Heap":111,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Berry":{"HeapUsed":4,"Objects":45},"Wifi":{"AP":1,"SSId":"Borealis","BSSId":"94:A6:7E:B1:72:2A","Channel":13,"Mode":"11n","RSSI":38,"Signal":-81,"LinkCount":1,"Downtime":"0T00:00:03"}}}

**Comment by msetina** [comment]
Board was flashed with tasmota32.factory.bin .
After reloading template commands return this:
09:58:12.380 CMD: Backlog Template; Module; GPIO 255
09:58:12.408 MQT: stat/small_tasmota32/RESULT = {"NAME":"8relays8switches","GPIO":[230,0,231,0,229,164,0,0,0,0,0,0,0,0,161,162,0,225,224,163,0,227,160,226,0,0,0,0,228,166,167,165,0,0,0,0],"FLAG":0,"BASE":1}
09:58:12.636 MQT: stat/small_tasmota32/RESULT = {"Module":{"0":"8relays8switches"}}
09:58:12.845 MQT: stat/small_tasmota32/RESULT = {"GPIO0":{"230":"Relay7"},"GPIO1":{"0":"None"},"GPIO2":{"231":"Relay8"},"GPIO3":{"0":"None"},"GPIO4":{"229":"Relay6"},"GPIO5":{"164":"Switch5"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"0":"None"},"GPIO14":{"0":"None"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"161":"Switch2"},"GPIO19":{"162":"Switch3"},"GPIO20":{"0":"None"},"GPIO21":{"225":"Relay2"},"GPIO22":{"224":"Relay1"},"GPIO23":{"163":"Switch4"},"GPIO24":{"0":"None"},"GPIO25":{"227":"Relay4"},"GPIO26":{"160":"Switch1"},"GPIO27":{"226":"Relay3"},"GPIO32":{"228":"Relay5"},"GPIO33":{"166":"Switch7"},"GPIO34":{"167":"Switch8"},"GPIO35":{"165":"Switch6"},"GPIO36":{"0":"None"},"GPIO37":{"0":"None"},"GPIO38":{"0":"None"},"GPIO39":{"0":"None"}}
09:58:30.381 CMD: Status0
09:58:30.420 MQT: stat/small_tasmota32/STATUS0 = {"Status":{"Module":0,"DeviceName":"ESP32 enota","FriendlyName":["rele1","rele2","rele3","rele4","rele5","rele6","rele7","rele8"],"Topic":"small_tasmota32","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0},"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32.bin","RestartReason":"Software reset CPU","Uptime":"0T00:01:27","StartupUTC":"2023-07-06T08:57:03","Sleep":50,"CfgHolder":4617,"BootCount":29,"BCResetTime":"2023-06-29T13:38:57","SaveCount":68},"StatusFWR":{"Version":"13.0.0(tasmota32)","BuildDateTime":"2023-06-26T10:54:38","Core":"2_0_10","SDK":"4.4.5.230614","CpuFrequency":160,"Hardware":"ESP32-D0WD-V3 rev.3","CR":"548/699"},"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Borealis",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000088","00006000","00004000","00000000"]},"StatusMEM":{"ProgramSize":1870,"Free":1009,"Heap":111,"StackLowMark":3,"PsrMax":0,"PsrFree":0,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"16405E","FlashFrequency":40,"FlashMode":"DIO","Features":["00000809","9F9AD7DF","0015A001","B7F7BFCF","05DA9BC0","E0360DC7","480840D2","20200000","D4BC482D","000A80B1"],"Drivers":"1,2,3,4,5,7,8,9,10,11,12,14,16,17,20,21,24,26,27,29,34,35,38,50,52,59,60,62,63,66,67,68,82,86,87,88","Sensors":"1,2,3,5,6,7,8,9,10,11,12,13,14,15,17,18,19,20,21,22,26,31,34,37,39,40,42,43,45,51,52,55,56,58,59,64,66,67,74,85,92,95,98,103,105,127","I2CDriver":"7,8,9,10,11,12,13,14,15,17,18,20,24,29,31,36,41,42,44,46,48,58,62,65,69,76,77"},"StatusNET":{"Hostname":"small-tasmota32-4864","IPAddress":"192.168.44.31","Gateway":"192.168.44.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.1","DNSServer2":"0.0.0.0","Mac":"7C:87:CE:30:33:00","IP6Global":"","IP6Local":"fe80::7e87:ceff:fe30:3300","Ethernet":{"Hostname":"","IPAddress":"0.0.0.0","Gateway":"0.0.0.0","Subnetmask":"0.0.0.0","DNSServer1":"192.168.1.1","DNSServer2":"0.0.0.0","Mac":"00:00:00:00:00:00","IP6Global":"","IP6Local":""},"Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":16.0},"StatusMQT":{"MqttHost":"192.168.1.66","MqttPort":1883,"MqttClientMask":"big_relay_box","MqttClient":"big_relay_box","MqttUser":"transmota","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4},"StatusTIM":{"UTC":"2023-07-06T08:58:30","Local":"2023-07-06T09:58:30","StartDST":"2023-03-26T02:00:00","EndDST":"2023-10-29T03:00:00","Timezone":"+01:00","Sunrise":"04:54","Sunset":"20:55"},"StatusSNS":{"Time":"2023-07-06T09:58:30","stikalo1":"ON","stikalo2":"ON","stikalo3":"ON","stikalo4":"ON","stikalo5":"ON","stikalo6":"OFF","stikalo7":"ON","stikalo8":"OFF"},"StatusSTS":{"Time":"2023-07-06T09:58:30","Uptime":"0T00:01:27","UptimeSec":87,"Heap":110,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Berry":{"HeapUsed":4,"Objects":45},"POWER1":"OFF","POWER2":"OFF","POWER3":"OFF","POWER4":"OFF","POWER5":"OFF","POWER6":"OFF","POWER7":"OFF","POWER8":"OFF","Wifi":{"AP":1,"SSId":"Borealis","BSSId":"94:A6:7E:B1:72:2A","Channel":13,"Mode":"11n","RSSI":44,"Signal":-78,"LinkCount":1,"Downtime":"0T00:00:02"}}}

**Comment by sfromis** [comment]
For readability, I'd suggest to wrap those copy/paste text blocks in triple backtick before and after.

Anyway the key thing is that the template is not activated in the first one, as `Module` being 0 means template activated, and thus with Module 1, not so. What you post does not give any indication of cause, or what any prior state even was. However, in case of boot loops (for whatever reason), Tasmota may "scale back" features until able to boot without crashing. This could include deactivating the template.

**Comment by msetina** [comment]
Thank you for pointing this out about the Module 1.
Restart from WebUI does not turn off template.
Neither does reset on the board.
First I will try to find a replicable procedure.

**Comment by msetina** [comment]
When template does not apply, this is on the top of log
```
00:00:00.003 HDW: ESP32-D0WD-V3 
00:00:00.043 UFS: FlashFS mounted with 304 kB free
00:00:00.065 CFG: Loaded from File, Count 77
00:00:00.067 FRC: Some settings have been reset (25)
00:00:00.086 TFS: File '.drvset003' not found
00:00:00.087 CFG: Energy use defaults as file system not ready or file not found
00:00:00.092 Project tasmota - ESP32 enota Version 13.0.0(tasmota32)-2_0_10(2023-06-26T10:54:38)

**Comment by msetina** [comment]
After an hour turned off, the board does a different boot, then at reset.
After a normal reset, this is at the top of log:
```
00:00:00.003 HDW: ESP32-D0WD-V3 
00:00:00.015 UFS: FlashFS mounted with 304 kB free
00:00:00.030 CFG: Loaded from File, Count 80
00:00:00.040 QPC: Count 1
00:00:00.125 BRY: Berry initialized, RAM used=4406 bytes
00:00:00.138 TFS: File '.drvset003' not found
00:00:00.138 CFG: Energy use defaults as file system not ready or file not found

**Comment by msetina** [comment]
A strange thing is that with only 1 minute without power, there is:
```00:00:00.003 HDW: ESP32-D0WD-V3 
00:00:00.032 UFS: FlashFS mounted with 304 kB free
00:00:00.051 CFG: Loaded from File, Count 82
00:00:00.053 FRC: Some settings have been reset (4)
00:00:00.080 TFS: File '.drvset003' not found
00:00:00.080 CFG: Energy use defaults as file system not ready or file not found
00:00:00.087 Project tasmota - ESP32 enota Version 13.0.0(tasmota32)-2_0_10(2023-06-26T10:54:38)

**Reply by msetina** [reply]
Since Berry is not reported to be initialized, the Berry scrpt console just prints entered commands

**Reply by sfromis** [reply]
Really bad: `FRC: Some settings have been reset (4)`
Even worse: `FRC: Some settings have been reset (25)`

Something bad happened, and Tasmota restarted, basically what I already mentioned.
You do not have old messages without a serial console connected.
One typical issue is not having a good power source, and engaged relays do draw quite a bit.

**Reply by msetina** [reply]
I am using onboard serial to USB that also provides power. If power should be the case for this board, I would probably need to power the board separately.

**Reply by msetina** [reply]
Will also check seral console.

**Reply by msetina** [reply]
Little progress on serial:
```00:00:00.019-179/41 UFS: FlashFS mounted with 304 kB free
00:00:00.031 CFG: Loaded from File, Count 88
00:00:00.033 FRC: Some settings have been reset (2)
00:00:00.120 BRY: Berry initialized, RAM used=4406 bytes
00:00:00.132 TFS: File '.drvset003' not found
00:00:00.132 CFG: Energy use defaults as file system not ready or file not found
00:00:00.146 Project tasmota - ESP32 enota Version 13.0.0(tasmota32)-2_0_10(2023-06-26T10:54:38)

Brownout detector was triggered

ets Jul 29 2019 12:21:46

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12732
ho 0 tail 12 room 4
load:0x40080400,len:2908
entry 0x400805c4

00:00:00.002 HDW: ESP32-D0WD-V3 
00:00:00.019 UFS: FlashFS mounted with 304 kB free
00:00:00.030 CFG: Loaded from File, Count 88
00:00:00.033 FRC: Some settings have been reset (3)
00:00:00.119 BRY: Berry initialized, RAM used=4406 bytes
00:00:00.131 TFS: File '.drvset003' not found

**Reply by msetina** [reply]
And my board is empty, just software.

**Reply by sfromis** [reply]
> Brownout detector was triggered

Translation: 
![image](https://github.com/arendst/Tasmota/assets/47082390/3345a27c-e40c-424e-b95c-26db61c51086)

**Reply by msetina** [reply]
Thank you for the guidance. I will need to do a proper power supply and serial console setup.  I hope it is not some regulator on the board issue, because I have connected the board as is to the 6W Power suply (it says 5V 1200mA) and it also showed some settings resets (3). Then I tried on my phone's PS which is 10W and I also got 7 resets. It is a Mini D1 ESP32 from AZ-delivery

**Reply by sfromis** [reply]
Oh crap. Those bad clones. Most likely has a paltry voltage regulator.

Of course, I don't _know_ what the issue is, whether it is a bad board, or a bad design, or your cable/power source is bad....

**Reply by msetina** [reply]
I tried to teach myself about this brownout issue. We can say, you do not have enough power, but as I learned it is the case that voltage drops under 2,44V in certain conditions. Since these are small boards there is no capacitors that would help mitigate this type of a problem. It looks as a capacitance problem in my case as it occurs after long off. It is seldom on reset.

**Reply by sfromis** [reply]
Right, a **lack** of capacitors increases the risk of "code brown" :hankey: with the ESP32 not getting the power it needs, at the right time.

**Reply by msetina** [reply]
It was a long time since I checked USB and to no surprise standard USB should not be able to suply enough power as it has 0,5A max (3.0 is 0,9A) . My board has min 0,5A in spec. So without proper power supply (nonstandard USB or dedicated supply) ESP32 should not get enough power.
So it is hard to live debug a fully populated ESP32 board just by connecting to PC.

**Reply by sfromis** [reply]
While standard USB is indeed max 0.5, it is not unexpected with ports only able to supply less. It *may* work better through a powered hub, assuming that the hub is "sufficiently transparent" to wrapped serial, which is not a certainty.

Another way is have an additional power source, common ground. However, feeding 5V from two places may not be safe, as that would assume all devices being designed for this, like backfeed diodes. Not very convenient with USB where you may not have much control over details.

One strategy could also be an external USB/serial adapter supplying TX/RX, a good power source for 5V, and common ground.

While capacitors can help, ideally boot should then be delayed till they are charged up, again not very convenient to set up.
